static/app/views/organizationContainer.tsx

```diff
@@ -1,21 +1,32 @@
-import styled from '@emotion/styled';
 import {useProfiler} from '@sentry/react';
 
+import {Container} from '@sentry/scraps/layout';
+
 import {Alert} from 'sentry/components/core/alert';
 import LoadingError from 'sentry/components/loadingError';
-import LoadingTriangle from 'sentry/components/loadingTriangle';
 import {ORGANIZATION_FETCH_ERROR_TYPES} from 'sentry/constants';
 import {t} from 'sentry/locale';
 import OrganizationStore from 'sentry/stores/organizationStore';
 import {useLegacyStore} from 'sentry/stores/useLegacyStore';
-import {space} from 'sentry/styles/space';
 
 function OrganizationLoadingIndicator() {
   /* Track how long this component is rendered for. */
   useProfiler('OrganizationLoadingIndicator', {
     hasRenderSpan: true,
   });
-  return <LoadingTriangle>{t('Loading data for your organization.')}</LoadingTriangle>;
+
+  /**
+   * This is the initial loader React will render as the app bootstraps.
+   * Rather than rendering a loader component, we can reuse the existing DOM
+   * provided by the server-rendered Django view!
+   *
+   * This ensures there are no layout shifts as the app initially boots up
+   * because the DOM is exactly the same and React doesn't have to reconcile
+   * the fallback state.
+   */
+  const ssrLoader = document.querySelector('#blk_router')?.innerHTML!;
+
+  return <div dangerouslySetInnerHTML={{__html: ssrLoader}} />;
```

Comment:
Does this cause any thrashing, or does react recognize that it's already rendered?

Reply:
No thrash! React knows the markup is the same.